backend/api/Controllers/RobotModelController.cs

```diff
@@ -0,0 +1,237 @@
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<RobotModel>> GetRobotModelById([FromRoute] string id)
+    {
+        var robotModel = await _robotModelService.ReadById(id);
+        if (robotModel == null)
+            return NotFound($"Could not find robotModel with id '{id}'");
+        return Ok(robotModel);
+    }
+
+    /// <summary>
+    /// Add a new robot model
+    /// </summary>
+    /// <remarks>
+    /// <para> This query adds a new robot model to the database </para>
+    /// </remarks>
+    [HttpPost]
+    [ProducesResponseType(typeof(RobotModel), StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<RobotModel>> Create(
+        [FromBody] CreateRobotModelQuery robotModelQuery
+    )
+    {
+        _logger.LogInformation("Creating new robot model");
+
+        RobotModel robotModel = new(robotModelQuery);
+
+        if (_robotModelService.ReadByRobotType(robotModel.Type).Result != null)
+            return BadRequest($"A robot already exists with the robot type '{robotModel.Type}");
+
+        try
+        {
+            var newRobotModel = await _robotModelService.Create(robotModel);
+            _logger.LogInformation(
+                "Successfully created new robot model with id '{robotModelId}'",
+                newRobotModel.Id
+            );
+            return CreatedAtAction(
+                nameof(GetRobotModelById),
+                new { id = newRobotModel.Id },
+                newRobotModel
+            );
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Error while creating new robot model");
+            throw;
+        }
+    }
+
+    /// <summary>
+    /// Updates a robot model in the database based on id
+    /// </summary>
+    /// <response code="200"> The robot model was successfully updated </response>
+    /// <response code="400"> The robot model data is invalid </response>
+    /// <response code="404"> There was no robot model with the given ID in the database </response>
+    [HttpPut]
+    [Route("{id}")]
+    [ProducesResponseType(typeof(RobotModel), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<Robot>> UpdateRobotModelById(
+        [FromRoute] string id,
+        [FromBody] UpdateRobotModelQuery robotModelQuery
+    )
```

Comment:
Would it be better if this was a patch endpoint?

Reply:
A patch would just updating one field, but here we are updating the entire model